src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionClassWriter.cs

```diff
@@ -21,7 +21,16 @@ public void ReflectionWriteClass(XmlWriterDelegator xmlWriter, object obj, XmlOb
         {
             InvokeOnSerializing(obj, context, classContract);
             obj = ResolveAdapterType(obj, classContract);
-            ReflectionWriteMembers(xmlWriter, obj, context, classContract, classContract, 0 /*childElementIndex*/, memberNames);
+            var iSerializableObj = obj as ISerializable;
```

Comment:
Yes, checking a bool is going to be cheaper than a cast which will need to traverse the type hierarchy to look for ISerializable in the type and parents.